### PR TITLE
Issue opl - passing requests/responses between WebRTC client and IMS

### PR DESCRIPTION
**Pull request by kacmos**
Feature flag act-as-ims-ua enables:
- registration of WebRTC clients is proxied to IMS,
- calls from/to WebRTC clients are passed to IMS,
- SIP requests to IMS has customized Call-ID and User-Agent headers,
- CDR's for calls between WebRTC clients and IMS are removed after call processing terminates,
- call hold actions for WebRTC clients and IMS are handled with the usage of SIP Message.

### CODE REVIEW

**Comment by gvagenas** [comment]
@kacmos Will review and update you here. Thanks

**Comment by gvagenas** [comment]
@kacmos please update your branch with latest master and resolve conflicts, thanks

**Comment by kacmos** [comment]
@gvagenas I've updated branch with latest master. 
I've added tests for IMS 200 OK delay configuration

**Comment by gvagenas** [comment]
Great, thanks @kacmos I will review asap

File: restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java [hunk 1]
```diff
@@ -1811,4 +1859,158 @@ private SipURI outboundInterface(String transport) {
...
+
+        if(registrations == null || registrations.size() ==0){
+            return null;
+        }
+        else{
+            return registrations.get(0);
+        }
+    }
+
+    private void imsProxyThroughMediaServer(final SipServletRequest request, final Client client,
```
Comment (gvagenas) [inline restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java]: @kacmos I think this method can be simplified more for better debugging later

File: restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java [hunk 2]
```diff
@@ -1811,4 +1859,158 @@ private SipURI outboundInterface(String transport) {
...
+            // We didn't find anyway to handle the call.
+            String errMsg = "Call cannot be processed because the registration: " + regUri.toString()
+                    + "cannot be found";
+            sendNotification(errMsg, 11005, "error", true);
+            return;
+        } else {
+            if (isFromIms) {
+                rcml = "<Response><Dial><Client>" + reg.getUserName() + "</Client></Dial></Response>";
+            }
+            boolean isFromWebRTC = false;
```
Comment (kacmos) [inline restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java]: @gvagenas 
you're right, this part of method is needless (it's a leftover from my work) - I remove it.

**Comment by kacmos** [comment]
@gvagenas I provided requested changes.
In case of feature description, I will provide it next week, together with description of #1716

**Comment by gvagenas** [comment]
Thanks @kacmos I will review and provide my comments, if any, as soon as possible. 
When documentation will be ready, I will merge with master

**Comment by gvagenas** [comment]
@kacmos any news on the documentation for this PR and the 200 OK Delay?

**Comment by kacmos** [comment]
@gvagenas 
I attach a document with description of this feature
[Issue opl - passing requests responses between WebRTC client and IMS.docx](https://github.com/RestComm/Restcomm-Connect/files/744532/Issue.opl.-.passing.requests.responses.between.WebRTC.client.and.IMS.docx)

**Comment by gvagenas** [comment]
Thanks @kacmos , I will review and update you if anything needed

**Pull request merged**